### PR TITLE
(GH-1) Initial version

### LINKED ISSUE #1: Basic implementation
Basic implementation:

* Support for creating code analysis reports from MsBuild log file

### PR DESCRIPTION
**Pull request by pascalberger**
Basic implementation of the addin.

Fixes #1

### CODE REVIEW

File: src/Cake.CodeAnalysisReporting/Reports/msbuild-xmlfilelogger-by-assembly.xsl [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+        background-color: transparent;
+        border: 0px;
+        }
+      </style>
+      <xsl:apply-templates select="$msbuild.root" />
+    </div>
+  </xsl:template>
+
+  <xsl:template match="build">
+    <!-- Remove .sln file from path to get the path of the solution file -->
```
Comment (pascalberger) [inline src/Cake.CodeAnalysisReporting/Reports/msbuild-xmlfilelogger-by-assembly.xsl:225]: This assumes a specific directory structure. Didn't had any idea for a more generic approach though

File: src/Cake.CodeAnalysisReporting/Reports/msbuild-xmlfilelogger-by-rule.xsl [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+        background-color: transparent;
+        border: 0px;
+        }
+      </style>
+      <xsl:apply-templates select="$compiler.root" />
+    </div>
+  </xsl:template>
+
+  <xsl:template match="build">
+    <!-- Remove .sln file from path to get the path of the solution file -->
```
Comment (pascalberger) [inline src/Cake.CodeAnalysisReporting/Reports/msbuild-xmlfilelogger-by-rule.xsl:240]: This assumes a specific directory structure. Didn't had any idea for a more generic approach though

File: src/Cake.CodeAnalysisReporting/CodeAnalysisReport.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    /// </summary>
+    public enum CodeAnalysisReport
+    {
+        /// <summary>
+        /// Undefined value.
+        /// </summary>
+        Undefined,
+
+        /// <summary>
+        /// Report from a MsBuild logfile genereated by the <code>XmlFileLogger</code> class from
```
Comment (stffabi) [inline src/Cake.CodeAnalysisReporting/CodeAnalysisReport.cs]: Typo: generated

File: src/Cake.CodeAnalysisReporting/CodeAnalysisReport.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        /// MSBuild Extension Pack grouped by rule number.
+        /// </summary>
+        MsBuildXmlFileLoggerByRule,
+
+        /// <summary>
+        /// Report from a MsBuild logfile genereated by the <code>XmlFileLogger</code> class from
```
Comment (stffabi) [inline src/Cake.CodeAnalysisReporting/CodeAnalysisReport.cs]: Typo: generated

**Review by stffabi — APPROVED** [review]
Some typos, the remaining LGTM.

File: src/Cake.CodeAnalysisReporting/CodeAnalysisReport.cs (hunk 3 above)
Comment (pascalberger) [reply src/Cake.CodeAnalysisReporting/CodeAnalysisReport.cs]: Fixed

File: src/Cake.CodeAnalysisReporting/Reports/msbuild-xmlfilelogger-by-assembly.xsl (hunk 1 above)
Comment (stffabi) [reply src/Cake.CodeAnalysisReporting/Reports/msbuild-xmlfilelogger-by-assembly.xsl:225]: Yeah, I know. Haven't found a more generic approach too for the first implementation we used so far.

File: src/Cake.CodeAnalysisReporting/CodeAnalysisReport.cs (hunk 4 above)
Comment (pascalberger) [reply src/Cake.CodeAnalysisReporting/CodeAnalysisReport.cs]: Fixed

**Pull request merged**